### ISSUE TITLE
Implement runtime support

### ISSUE THREAD

**Issue by DmitrySoshnikov** [DmitrySoshnikov/babel-plugin-transform-modern-regexp#3]
> NOTE: this is mostly for spec-compatibility; practically regexes can be used without runtime 

As the comments says, a transform [can accept](https://github.com/DmitrySoshnikov/babel-plugin-transform-modern-regexp/blob/master/index.js#L32) `includeRuntime` option (which is currently [disabled](https://github.com/DmitrySoshnikov/babel-plugin-transform-modern-regexp/blob/master/index.js#L61-L63)).

If the option is passed, regexes are wrapped into `RegExpTree` wrapper, and provide extra functionality at runtime (e.g. accessing captured named groups on matched result).

The runtime implementation [already exists](https://github.com/DmitrySoshnikov/regexp-tree/blob/master/src/compat-transpiler/runtime/index.js), we need to pull it into a separate package, which can be auto-required by the transform.

```js
const re = new RegExp(`

  # A regular expression for date.

  (?<year>\d{4})-    # year part of a date
  (?<month>\d{2})-   # month part of a date
  (?<day>\d{2})      # day part of a date

`, 'x');

const result = re.exec('2017-04-17');

console.log(result.groups.year); // 2017
```

Is translated into:

```js
// Auto-require `regexp-tree-runtime`, it should be in the dependencies.

const RegExpTree = require('regexp-tree-runtime');

...

const re = new RegExpTree(/(\d{4})-(\d{2})-(\d{2})/, {
  flags: 'x',
  source: <original-source>,
  groups: {
    year: 1,
    month: 2,
    day: 3,
  },
});

const result = re.exec('2017-04-17');

console.log(result.groups.year); // 2017
```

**Comment by mike-marcacci** [comment]
Is this still the latest on this issue? Named capture groups would be a game changer for me, and I can try to find time to work on this if it's still the preferred strategy.

**Comment by DmitrySoshnikov** [comment]
@mike-marcacci, yes, would be great to add this, and I agree, actual usage of the groups is a game changer. We can discuss any strategy here, those described above are just an example, I'm happy to consider any alternatives you have. And of course I'll appreciate a PR.

CC @nicolo-ribaudo who was trying to add a PR to Babel, but the https://github.com/babel/babel/pull/7105 PR itself seems stuck there forever, so we can add it here.

**Comment by nicolo-ribaudo** [comment]
Yeah sorry, currently I'm working on decorators which are a more requested feature. I will revisit that PR after them.
